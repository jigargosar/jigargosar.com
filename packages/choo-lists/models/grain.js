const R = require('ramda')
const nanoid = require('nanoid')

const faker = require('faker')
faker.seed(123)

export function createNew({text} = {}) {
  return {
    id: `grain-${nanoid()}`,
    text: R.isNil(text) ? `${faker.lorem.words()}` : text,
  }
}

export function updateText(text, grain) {
  return R.assoc('text', text, grain)
}

export function text(grain) {
  return grain.text
}

export function id(grain) {
  return grain.id
}
