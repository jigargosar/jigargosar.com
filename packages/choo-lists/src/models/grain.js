const R = require('ramda')
const RA = require('ramda-adjunct')
const nanoid = require('nanoid')
const assert = require('assert')

const idPropName = '_id'
const deletedPropName = '_deleted'
const revisionPropName = '_rev'
const versionPropName = 'version'
const textPropName = 'text'
const createdAtPropName = 'createdAt'

export function createNew({text = ''} = {}) {
  const nowTimestamp = Date.now()
  assert(RA.isNotNil(text))
  return {
    [idPropName]: `grain-${nowTimestamp}-${nanoid()}`,
    text,
    [createdAtPropName]: nowTimestamp,
    [versionPropName]: 0,
  }
}

export function setText(text, grain) {
  assert(RA.isString(text))
  return R.assoc(textPropName, text, grain)
}

export function getText(grain) {
  return grain[textPropName]
}

export function getId(grain) {
  return grain[idPropName]
}

export function validate(doc) {
  assert(RA.isString(doc[idPropName]))
  assert(RA.isString(doc[revisionPropName]))
  assert(RA.isString(doc[textPropName]))
  assert(RA.isNumber(doc[createdAtPropName]))
  assert(RA.isNumber(doc[versionPropName]))
  return doc
}

export const idEq = R.curry(function idEq(id, grain) {
  return R.propEq(idPropName, id, grain)
})

export const eqById = R.curry(function idEq(grain1, grain2) {
  return R.eqProps(idPropName, grain1, grain2)
})
