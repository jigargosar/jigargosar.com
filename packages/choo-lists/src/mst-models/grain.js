import {BaseModel} from './base-model'
import {types} from 'mobx-state-tree'

const R = require('ramda')
const RA = require('ramda-adjunct')

const assert = require('assert')

export const Grain = BaseModel.named('Grain')
  .props({
    text: types.optional(types.string, () => ''),
  })
  .views(self => ({
    getText() {
      return self.text
    },
    getUserEditableProps() {
      return ['text']
    },
  }))

export const GrainStore = types
  .model('GrainStore', {
    modelMap: types.optional(types.map(Grain), () => ({})),
  })
  .views(self => ({
    getList() {
      return Array.from(self.modelMap.values())
    },
  }))
  .actions(self => ({
    add(model) {
      self.modelMap.put(model)
    },
  }))
