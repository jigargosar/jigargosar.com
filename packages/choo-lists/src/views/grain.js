const {updateTitle} = require('./events')
const {CodeMirrorEditor} = require('../components/CodeMirrorEditor')
const R = require('ramda')
const log = require('nanologger')('views:list')
const html = require('choo/html')
const {TITLE} = require('./meta')
const EM = require('../models/edit-mode')
const domAutofocus = require('dom-autofocus')
const yaml = require('js-yaml')
const {actions: GEA} = require('../stores/edit-grain-store')

module.exports = view

const centeredContentClass = 'center mw7 mv3 ph3'

function view(state, emit) {
  updateTitle(TITLE, state, emit)
  return html`<body class="sans-serif lh-copy f4">
      <div class="bg-light-blue tc pa3">
        <div class="f1">${TITLE}</div>
      </div>
      <div 
        class="${centeredContentClass}" 
        style="min-width: 100%;">View Grain</div>
    </body>`
}
