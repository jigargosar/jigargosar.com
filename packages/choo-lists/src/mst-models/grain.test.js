import {Grain} from './grain'

const log = require('nanologger')('grain.test')


// noinspection SpellCheckingInspection
jest.mock('nanoid', () => jest.fn(() => 'BitYkpcz~i4qTWDlMgGbS'))
jest.spyOn(Date, "now").mockImplementation(() => 1527958482904)

beforeEach(() => {
  jest.resetModules()
})

describe('Grain', function() {
  test('should get created without any arguments', function() {

    // Date.now = jest.fn(() => 1527958482904)
    log.debug(Grain.create())
    expect(Grain.create()).toMatchSnapshot()
  })
})
