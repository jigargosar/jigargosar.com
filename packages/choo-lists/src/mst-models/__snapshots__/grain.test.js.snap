// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grain should get created without any arguments 1`] = `
Object {
  "0": "G",
  "1": "r",
  "2": "a",
  "3": "i",
  "4": "n",
  "createdAt": 1527958482904,
  "id": "BitYkpcz~i4qTWDlMgGbS",
  "modifiedAt": 1527958482904,
}
`;
